src/settings/components/SettingForm.tsx

```diff
@@ -1,50 +1,148 @@
+import { useRef, useState, useEffect } from "react";
+import { useTypedSelector } from "src/store";
+import { SettingSchema, SettingSchemaType } from "src/settings/validations";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { FormControlContainer } from "src/shared/components";
+import avatarPlaceholder from "src/assets/images/avatar-placeholder.png";
+import { updateUser } from "src/store/auth";
+import { useDispatch } from "react-redux";
+import { AppDispatch } from "src/store/types";
+import { getFormDataFromObject } from "src/helpers/form-data-helper";
+import { convertImageToBase64 } from "src/helpers/convert-image";
+import { useMessageToast } from "src/hooks";
 
 export const SettingForm = () => {
+  const { user, errorMessage, successMessage, isLoading } = useTypedSelector(
+    ({ auth }) => auth
+  );
+
+  const { displaySuccessMessage, displayError } = useMessageToast();
+
+  const [userAvatar, setUserAvatar] = useState<string | ArrayBuffer | null>(
+    user?.avatar ? (user?.avatar as string) : null
+  );
+
+  const dispatch = useDispatch<AppDispatch>();
+
   const profileImageInputRef = useRef<HTMLInputElement | null>(null);
+
   const onUploadImage = (): void => {
     profileImageInputRef.current?.click();
   };
 
+  const formDefaultValues = {
+    ...user,
+    password: "",
+  };
+  const {
+    reset,
+    setValue,
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm<SettingSchemaType>({
+    defaultValues: formDefaultValues,
+    resolver: zodResolver(SettingSchema),
+  });
+
+  const onSubmitForm: SubmitHandler<SettingSchemaType> = (data) => {
+    const userData = getFormDataFromObject(data);
+    dispatch(
+      updateUser({
+        id: user?._id as string,
+        userData: userData,
+      })
+    );
+  };
+  useEffect(() => {
+    if (successMessage) {
+      displaySuccessMessage(successMessage);
+    } else if (errorMessage) {
+      displayError(errorMessage);
+    }
+  }, [errorMessage, displayError, displaySuccessMessage, successMessage]);
+
   return (
-    <div className="w-full h-screen bg-[#D3D3D3] text-[#2F2F2F]">
+    <div className="w-full h-screen bg-secondary-200 text-primary-500">
       <div className="flex flex-col items-center justify-center h-screen">
         <Box>
           <Image
             content="center"
             borderRadius="full"
             boxSize="200px"
-            alt="Dan Abramov"
+            alt="avatar"
             cursor="pointer"
             onClick={onUploadImage}
-            src="https://bit.ly/dan-abramov"
+            src={userAvatar ? (userAvatar as string) : avatarPlaceholder}
           />
           <Input
+            onChange={({ target: { files } }) => {
+              setValue("avatar", files?.[0]);
+              if (files?.[0]) {
+                convertImageToBase64(files?.[0], setUserAvatar);
+              }
+            }}
```

Comment:
```suggestion
            onChange={({ target: { files } }) => {
              const avatarFile = files?.[0];
              setValue("avatar", avatarFile);
              if (avatarFile) {
                convertImageToBase64(avatarFile, setUserAvatar);
              }
            }}
```
Se podría crear una constante y mover todo a otra función `onUploadAvatar`
```
onChange={onUploadAvatar}
```